### PR TITLE
sql: disallow SHOW TRANSACTION ISOLATION/PRIORITY outside of txn

### LINKED ISSUE #6151: sql: disallow setting and viewing the txn iso level outside of a txn
Currently, you can set the isolation level for a specific transaction within a transaction block, but you can't set the isolation level for an entire session.

**Failing to set isolation level for session:**

```
JESSEs-MacBook-Pro:cockroach jesseseldess$ ./cockroach sql
# Welcome to the cockroach SQL interface.
# All statements must be terminated by a semicolon.
# To exit: CTRL + D.
root@:26257> show transaction isolation level;
+-----------------------------+
| TRANSACTION ISOLATION LEVEL |
+-----------------------------+
| SERIALIZABLE                |
+-----------------------------+
root@:26257> set transaction isolation level snapshot;
pq: there is no transaction in progress
root@:26257> set session transaction isolation level snapshot;
pq: there is no transaction in progress
root@:26257> set local transaction isolation level snapshot;
pq: there is no transaction in progress
```

**Successfully setting isolation level for a specific transaction:**

```
root@:26257> begin;
BEGIN
root@:26257> set transaction isolation level snapshot;
SET TRANSACTION
root@:26257> show transaction isolation level;
+-----------------------------+
| TRANSACTION ISOLATION LEVEL |
+-----------------------------+
| SNAPSHOT                    |
+-----------------------------+
root@:26257> commit;
COMMIT
root@:26257> show transaction isolation level;
+-----------------------------+
| TRANSACTION ISOLATION LEVEL |
+-----------------------------+
| SERIALIZABLE                |
+-----------------------------+
```

### PR DESCRIPTION
**Pull request by spencerkimball**
Fixes #6151

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/14491)

### CODE REVIEW

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[pkg/sql/show.go, line 60 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491#-KgWDmcSTGq8tjvqiMDo:-KgWDmcSTGq8tjvqiMDp:b91pig1) ([raw file](https://github.com/cockroachdb/cockroach/blob/527cbcedc118e8b7f2947950be51b1f44a5cb720/pkg/sql/show.go#L60)):*
> ```Go
> }()
> var varRequireExplicitTxn = map[string]struct{}{
> 	`transaction isolation level`: struct{}{},
> ```

Can you avoid duplicating these with the other location in which these verbatim strings are used?

---

*[pkg/sql/testdata/logic_test/txn, line 414 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491#-KgWE1fCmin68FCqH4Q3:-KgWE1fCmin68FCqH4Q4:bmcpzn2) ([raw file](https://github.com/cockroachdb/cockroach/blob/527cbcedc118e8b7f2947950be51b1f44a5cb720/pkg/sql/testdata/logic_test/txn#L414)):*
> ```
> SNAPSHOT
> 
> # It is an error to show the isolation level without a transaction.
> ```

i think you can just remove this comment.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491#-:-KgWE3Yjg04fnP6KwMgm:bnfp4nl)*

File: pkg/sql/show.go [hunk 1]
```diff
@@ -56,6 +56,10 @@ var varNames = func() []string {
 	sort.Strings(res)
 	return res
 }()
+var varRequireExplicitTxn = map[string]struct{}{
```
Comment (knz) [inline pkg/sql/show.go]: Make `varGen` above a `map[string]struct{ bool, func }` to embed this information there.

**Comment by spencerkimball** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[pkg/sql/show.go, line 59 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491#-KgWC8a0dHOUUFDhjL7c-r1-59:-KgZXw8VNcXcsxDAiuVs:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/527cbcedc118e8b7f2947950be51b1f44a5cb720/pkg/sql/show.go#L59)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Make `varGen` above a `map[string]struct{ bool, func }` to embed this information there.
</blockquote></details>

Done.

---

*[pkg/sql/show.go, line 60 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491#-KgWDmcSTGq8tjvqiMDo:-KgZXwtvvJepzJYw6_d7:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/527cbcedc118e8b7f2947950be51b1f44a5cb720/pkg/sql/show.go#L60)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

Can you avoid duplicating these with the other location in which these verbatim strings are used?
</blockquote></details>

Done.

---

*[pkg/sql/testdata/logic_test/txn, line 414 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491#-KgWE1fCmin68FCqH4Q3:-KgZY0BY6VumSzCv3Qmv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/527cbcedc118e8b7f2947950be51b1f44a5cb720/pkg/sql/testdata/logic_test/txn#L414)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

i think you can just remove this comment.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491)*

**Comment by tamird** [comment]
What about setting/resetting these variables outside a transaction?

---

Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491#-:-KgZyXvthCBFSrkj_Bew:byux18t)*

**Comment by spencerkimball** [comment]
That was already disallowed.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/14491#-:-Kg_3Mf_55hoHRe9huCO:b-2nxglb)*

**Review by knz — COMMENTED** [review]
Thanks!

**Pull request merged**